meta/meta_autoid.go

```diff
@@ -0,0 +1,205 @@
+		return 0, errors.Trace(err)
+	}
+	// Check if table exists.
+	tableKey := m.tableKey(a.tableID)
+	if err := m.checkTableExists(dbKey, tableKey); err != nil {
+		return 0, errors.Trace(err)
+	}
+
+	return m.txn.HInc(dbKey, a.tableIDEncFn(a.tableID), step)
+}
+
+// Del implements the interface AutoIDAccessor.
+func (a autoIDAccessor) Del() error {
+	m := a.m
+	dbKey := m.dbKey(a.databaseID)
+	if err := m.txn.HDel(dbKey, a.tableIDEncFn(a.tableID)); err != nil {
+		return errors.Trace(err)
+	}
+	return nil
+}
+
+var _ AutoIDAccessors = autoIDAccessors{}
+
+// AutoIDAccessors represents all the auto IDs of a table.
+type AutoIDAccessors interface {
+	Get() (AutoIDs, error)
+	Put(autoIDs AutoIDs) error
+	Del() error
+}
+
+type autoIDAccessors struct {
+	rowIDAcc  AutoIDAccessor
+	randIDAcc AutoIDAccessor
+}
+
+// Get implements the interface AutoIDAccessors.
+func (a autoIDAccessors) Get() (autoIDs AutoIDs, err error) {
+	autoIDs.RowID, err = a.rowIDAcc.Get()
+	if err != nil {
+		return
+	}
+	autoIDs.RandomID, err = a.randIDAcc.Get()
+	if err != nil {
+		return
+	}
+	return
+}
+
+// Put implements the interface AutoIDAccessors.
+func (a autoIDAccessors) Put(autoIDs AutoIDs) error {
+	if err := a.rowIDAcc.Put(autoIDs.RowID); err != nil {
+		return err
+	}
+	return a.randIDAcc.Put(autoIDs.RandomID)
+}
+
+// Del implements the interface AutoIDAccessors.
+func (a autoIDAccessors) Del() error {
+	if err := a.rowIDAcc.Del(); err != nil {
+		return err
+	}
+	return a.randIDAcc.Del()
+}
+
+// AutoIDCtrl can be changed into rowid/auto_increment/auto_random ID accessor.
+type AutoIDCtrl struct {
+	access autoIDAccessor
+}
+
+// RowID is used to get the _tidb_rowid meta key-value accessor.
+func (a *AutoIDCtrl) RowID() AutoIDAccessor {
+	a.access.tableIDEncFn = a.access.m.autoTableIDKey
+	return a.access
+}
+
+// IncrementID is used to get the auto_increment ID meta key-value accessor.
+func (a *AutoIDCtrl) IncrementID() AutoIDAccessor {
+	a.access.tableIDEncFn = a.access.m.autoIncrementIDKey
+	return a.access
+}
+
+// RandomID is used to get the auto_random ID meta key-value accessor.
+func (a *AutoIDCtrl) RandomID() AutoIDAccessor {
+	a.access.tableIDEncFn = a.access.m.autoRandomTableIDKey
+	return a.access
+}
+
+// All is used to get all auto IDs meta key-value accessor.
+func (a *AutoIDCtrl) All() AutoIDAccessors {
+	acc := autoIDAccessor{
+		m:          a.access.m,
+		databaseID: a.access.databaseID,
+		tableID:    a.access.tableID,
+	}
+	randAcc := acc
+	acc.tableIDEncFn = a.access.m.autoTableIDKey
+	randAcc.tableIDEncFn = a.access.m.autoRandomTableIDKey
+	return autoIDAccessors{
+		rowIDAcc:  acc,
+		randIDAcc: randAcc,
+	}
+}
+
+// NewAutoIDCtrl creates a new AutoIDCtrl.
+func NewAutoIDCtrl(m *Meta, databaseID, tableID int64) *AutoIDCtrl {
+	return &AutoIDCtrl{
+		autoIDAccessor{
+			m:          m,
+			databaseID: databaseID,
+			tableID:    tableID,
+		},
+	}
+}
+
+// AutoIDs represents all auto IDs of a table.
+type AutoIDs struct {
```

Comment:
s/AutoIDs/AutoIDPair

Reply:
But the word "Pair" means 2. Later I will add another ID named `auto_increment`.